### PR TITLE
Update Publish workflow to be manually run

### PR DESCRIPTION
**Pull request by HettieS**
Prevent Publish gem workflow from running in CI. I'm a control freak who doesn't want to live life on the edge, so let's run it manually.

### CODE REVIEW

File: .github/workflows/publish_gem.yml [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Tag
+        required: true
 jobs:
   build:
+    name: Publish to RubyGems.org and GitHub Packages
     runs-on: ubuntu-latest
     if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (HettieS) [inline .github/workflows/publish_gem.yml:12]: not 100% sure if I need this line

File: .github/workflows/publish_gem.yml [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
-        fi
-        if ! git ls-remote --tags --exit-code origin v${CURRENT_VERSION}; then
-          git tag v${CURRENT_VERSION}
-          git push --tags
-        fi
+    - name: Publish gem
+      uses: dawidd6/action-publish-gem@v1
+      with: 
+        api_key: ${{ secrets.RUBYGEMS_API_KEY }}
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (peteryates) [inline .github/workflows/publish_gem.yml:33]: Not sure you'd need the `GITHUB_TOKEN` either if you're just pushing to Rubygems.

File: .github/workflows/publish_gem.yml [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
-  - push
-
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Tag
+        required: true
 jobs:
   build:
+    name: Publish to RubyGems.org and GitHub Packages
```
Comment (peteryates) [inline .github/workflows/publish_gem.yml:10]:
```suggestion
    name: Publish to RubyGems.org
```

File: .github/workflows/publish_gem.yml [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
-    - uses: ruby/setup-ruby@v1
-    - name: Install dependencies
-      run: gem update --system
+      with:
+        ref: ${{ github.event.inputs.tag }}
+
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '2.7.5'
```
Comment (peteryates) [inline .github/workflows/publish_gem.yml:20]: I'd probably go with '2.7.6' now as it's the latest in the 2.7 series

**Review by peteryates — APPROVED** [review]

**Pull request merged**